ts/features/design-system/core/DSIcons.tsx

```diff
@@ -4,26 +4,43 @@ import { DSIconViewerBox, iconItemGutter } from "../components/DSIconViewerBox";
 import {
   Icon,
   IOIcons,
-  IOIconType,
-  IconNav,
   IONavIcons,
-  IONavIconType,
-  IconBiometric,
   IOBiometricIcons,
-  IOBiometricIconType,
-  IconCategory,
   IOCategoryIcons,
-  IOCategoryIconType,
-  IconProduct,
   IOProductIcons,
-  IOProductIconType,
-  IOIconsNew
+  IOIconsNew,
+  SVGIconProps
 } from "../../../components/core/icons";
 import { H2 } from "../../../components/core/typography/H2";
 import { H3 } from "../../../components/core/typography/H3";
 import type { IOColors } from "../../../components/core/variables/IOColors";
 import { DesignSystemScreen } from "../components/DesignSystemScreen";
 
+// Filter the main object, removing already displayed icons in the other sets
+type IconSubsetObject = Record<
+  string,
+  ({ size, style }: SVGIconProps) => JSX.Element
+>;
+interface IconSetObject {
+  [key: string]: ({ size, style }: SVGIconProps) => JSX.Element;
+}
+const filterIconSet = (
+  iconSubsetObject: IconSubsetObject,
+  iconSetObject: IconSetObject
+): IconSetObject => {
+  const iconSetArray = Object.keys(iconSubsetObject);
+  return Object.fromEntries(
+    Object.entries(iconSetObject).filter(([key]) => !iconSetArray.includes(key))
+  );
+};
```

Comment:
This should be more functional using [fp-ts Record](https://gcanti.github.io/fp-ts/modules/Record.ts.html). What do you think about?
```suggestion
const filterIconSet = (
  iconSubsetObject: IconSubsetObject,
  iconSetObject: IconSetObject
): IconSetObject => pipe(
  iconSetObject,
  toEntries,
  entries => entries.filter(([key]) => !Object.keys(iconSubsetObject).includes(key)),
  fromEntries
);
```

Reply:
It could be a great replacement, but it changes the original elements' order. We'd like to keep it because we like to group the various icons with conceptually similar ones.